main.go

```diff
@@ -171,17 +198,20 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 		}
 	}
 }
-
 func main() {
-	runtime.GOMAXPROCS(2)
-	if len(os.Args) != 2 {
-		log.Fatalf("Usage: %s METRICS_URL", os.Args[0])
-	}
-
+	runtime.GOMAXPROCS(4)
+	url := flag.String("url", "https://example.com", "url")
+	cert := flag.String("cert", "/tmp/admin.crt", "certificate file")
+	key := flag.String("key", "/tmp/admin.key", "key file")
+	flag.Parse()
 	mfChan := make(chan *dto.MetricFamily, 1024)
-
-	go fetchMetricFamilies(os.Args[1], mfChan)
-
+	if flag.NFlag() == 1 {
```

Comment:
Here, checking number of flags would be deceptive if news flags would be added.  Instead, if the flags are present, and values for *both* `key` and `cert` were provided, then use `fetchMetricFamiliesWithAuth`, otherwise use `fetchMetricFamilies`.